### ISSUE TITLE
Support a fixed-point buiild

### ISSUE THREAD

**Issue by jwcullen** [AOMediaCodec/libiamf#106]
It's common for this type of software to be configurable to work better on fixed-point systems.

  - https://github.com/xiph/opus/blob/main/README#L166-L171

There may be other parts that are floating point, but at least demixing module and resampler only use floating point.

I think this could be a big task to fix, but it may make the software more useful for use on certain systems.

**Comment by yilun-zhangs** [comment]
@jwcullen 
Yes, it may require fixed-point version reference for certain systems, 
and this is a big task to support fixed-point option, we will enhance this next, it may take long time to proceed.

**Comment by yilun-zhangs** [comment]
Hi @jwcullen 
Do you know how many bits dose libopus fixed-point reference support?
If we want libiamf to be configurable to work better on fixed-point systems, are there any requirement to the bits of fixed-point?

**Comment by jwcullen** [comment]
> Do you know how many bits dose libopus fixed-point reference support?

Before version 1.5: 16-bit.
Version 1.5+: 16 or [new 24-bit](https://github.com/xiph/opus/commit/977ee01670a9e5271ffeeddf5597583458002dfd).

> If we want libiamf to be configurable to work better on fixed-point systems, are there any requirement to the bits of fixed-point?

I filed this issue based on vendor feedback. I do not have specific requirements.

I believe it is understood that a fixed point configuration is intended to be faster on embedded systems without an [FPU](https://en.wikipedia.org/wiki/Floating-point_unit), at the cost of (potentially) reduced audio quality.
